src/fsharp/FSharp.Core/map.fs

```diff
@@ -598,6 +613,56 @@ namespace Microsoft.FSharp.Collections
             member s.TryGetValue(k,r) = if s.ContainsKey(k) then (r <- s.[k]; true) else false
             member s.Remove(k : 'Key) = ignore(k); (raise (NotSupportedException(SR.GetString(SR.mapCannotBeMutated))) : bool)
 
+        interface System.Collections.IDictionary with
+            member m.Item
+                with get key =
+                    match key with
+                    | null ->
+                        // According to the documentation for IDictionary, implementations should
+                        // raise ArgumentNullException if the key passed to the indexer is null.
+                        nullArg "key"
+                    | :? 'Key as key ->
+                        match m.TryFind key with
+                        | Some v -> box v
+                        | None -> null
+                    | _ -> null
+                and  set key value =
```

Comment:
@KevinRansom considering the class already implements IDictionary<K,V> the same way (throwing on set value, etc.) I think this is reasonable addition, most people will know that passing a map to a place expecting a dictionary, only lookup / enumeration scenarios would be expected to work.

I've been facing situations where a type would implement IDictionary<K,V> but I'd actually needed the older interface to pass to some code, and doing the wrapper myself felt absurd.

The IReadOnly* stuff sadly came late in the game, it will take time for it to be used more readily where it should be used.